File: mambo-protocol/src/main/scala/org/mambo/protocol/types/game/house/HouseInformationsForGuild.scala





















// Generated on 05/03/2013 10:35:07
// Scala version by Blackrush
package org.mambo.protocol.types

import org.mambo.core.io._
import org.mambo.protocol.{Type => BaseType, _}
import org.mambo.protocol.enums._

trait HouseInformationsForGuild extends TypeDeserializer {
  val typeId = 170

  protected trait Type extends BaseType with Serializable {
    val typeId = HouseInformationsForGuild.typeId

    val houseId: Int
    val modelId: Int
    val ownerName: String
    val worldX: Short
    val worldY: Short
    val mapId: Int
    val subAreaId: Short
    val skillListIds: Seq[Int]
    val guildshareParams: UInt

    def serialize(buf: Buffer) {
      buf.writeInt(houseId)
      buf.writeInt(modelId)
      buf.writeString(ownerName)
      buf.writeShort(worldX)
      buf.writeShort(worldY)
      buf.writeInt(mapId)
      buf.writeShort(subAreaId)
      buf.writeUShort(skillListIds.length)
      for (entry <- skillListIds) {
        buf.writeInt(entry)
      }
      buf.writeUInt(guildshareParams)
    }

  }

  protected def create(houseId: Int, modelId: Int, ownerName: String, worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, skillListIds: Seq[Int], guildshareParams: UInt): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val houseId = buf.readInt()
    val modelId = buf.readInt()
    val ownerName = buf.readString()
    val worldX = buf.readShort()
    val worldY = buf.readShort()
    val mapId = buf.readInt()
    val subAreaId = buf.readShort()
    val skillListIds = {
      val builder = collection.immutable.Seq.newBuilder[Int]
      for (i <- (0 to buf.readUShort())) {
        builder += buf.readInt()
      }
      builder.result()
    }
    val guildshareParams = buf.readUInt()

    create(houseId, modelId, ownerName, worldX, worldY, mapId, subAreaId, skillListIds, guildshareParams)
  }

}

object HouseInformationsForGuild extends HouseInformationsForGuild {
  case class TypeImpl(houseId: Int, modelId: Int, ownerName: String, worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, skillListIds: Seq[Int], guildshareParams: UInt) extends Type
  override def create(houseId: Int, modelId: Int, ownerName: String, worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, skillListIds: Seq[Int], guildshareParams: UInt) = TypeImpl(houseId, modelId, ownerName, worldX, worldY, mapId, subAreaId, skillListIds, guildshareParams)

  def unapply(msg: TypeImpl) = TypeImpl.unapply(msg)
  def apply(houseId: Int, modelId: Int, ownerName: String, worldX: Short, worldY: Short, mapId: Int, subAreaId: Short, skillListIds: Seq[Int], guildshareParams: UInt) = create(houseId, modelId, ownerName, worldX, worldY, mapId, subAreaId, skillListIds, guildshareParams)

  type Target = TypeImpl

}

File: mambo-protocol/src/main/java/org/mambo/protocol/types/HouseInformationsForGuild.java


// Generated on 05/08/2013 19:38:02
package org.mambo.protocol.types;

import java.util.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class HouseInformationsForGuild extends NetworkType {
    public static final short TYPE_ID = 170;
    
    @Override
    public short getTypeId() {
        return TYPE_ID;
    }
    
    public int houseId;
    public int modelId;
    public String ownerName;
    public short worldX;
    public short worldY;
    public int mapId;
    public short subAreaId;
    public int[] skillListIds;
    public long guildshareParams;
    
    public HouseInformationsForGuild() { }
    
    public HouseInformationsForGuild(int houseId, int modelId, String ownerName, short worldX, short worldY, int mapId, short subAreaId, int[] skillListIds, long guildshareParams) {
        this.houseId = houseId;
        this.modelId = modelId;
        this.ownerName = ownerName;
        this.worldX = worldX;
        this.worldY = worldY;
        this.mapId = mapId;
        this.subAreaId = subAreaId;
        this.skillListIds = skillListIds;
        this.guildshareParams = guildshareParams;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeInt(houseId);
        buf.writeInt(modelId);
        buf.writeString(ownerName);
        buf.writeShort(worldX);
        buf.writeShort(worldY);
        buf.writeInt(mapId);
        buf.writeShort(subAreaId);
        buf.writeUShort(skillListIds.length);
        for (int entry : skillListIds) {
            buf.writeInt(entry);
        }
        buf.writeUInt(guildshareParams);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        houseId = buf.readInt();
        if (houseId < 0)
            throw new RuntimeException("Forbidden value on houseId = " + houseId + ", it doesn't respect the following condition : houseId < 0");
        modelId = buf.readInt();
        if (modelId < 0)
            throw new RuntimeException("Forbidden value on modelId = " + modelId + ", it doesn't respect the following condition : modelId < 0");
        ownerName = buf.readString();
        worldX = buf.readShort();
        if (worldX < -255 || worldX > 255)
            throw new RuntimeException("Forbidden value on worldX = " + worldX + ", it doesn't respect the following condition : worldX < -255 || worldX > 255");
        worldY = buf.readShort();
        if (worldY < -255 || worldY > 255)
            throw new RuntimeException("Forbidden value on worldY = " + worldY + ", it doesn't respect the following condition : worldY < -255 || worldY > 255");
        mapId = buf.readInt();
        subAreaId = buf.readShort();
        if (subAreaId < 0)
            throw new RuntimeException("Forbidden value on subAreaId = " + subAreaId + ", it doesn't respect the following condition : subAreaId < 0");
        int limit = buf.readUShort();
        skillListIds = new int[limit];
        for (int i = 0; i < limit; i++) {
            skillListIds[i] = buf.readInt();
        }
        guildshareParams = buf.readUInt();
        if (guildshareParams < 0 || guildshareParams > 4294967295L)
            throw new RuntimeException("Forbidden value on guildshareParams = " + guildshareParams + ", it doesn't respect the following condition : guildshareParams < 0 || guildshareParams > 4294967295L");
    }
    
}
